Create screens

gHashTag opened the pull request:
- fix qr scanning bug
- swipeable row closes when the item changes
- fix android input style and addition style for disabled input
- KeyboardAvoidingView fix android
- useTypedNavigation hook

#304

vivalaakam commented on src/hooks/useTypedNavigation.ts:
```diff
@@ -0,0 +1,6 @@
+import {useNavigation} from '@react-navigation/native';
```
use kebab-case pls

vivalaakam commented on src/screens/settings-account-detail.tsx:
```diff
@@ -31,29 +30,8 @@ export const SettingsAccountDetailScreen = () => {
-      },
-    );
-  }, [wallet]);
+    navigation.navigate('editAccountName', route.params);
```
pls rename screen to settingsAccountEdit

vivalaakam commented on src/screens/edit-contact.tsx:
```diff
@@ -0,0 +1,84 @@
+import React, {useState} from 'react';
```
rename pls to transaction-contact-edit

vivalaakam commented:
Instead hooks I suggest to make it as transaction sum, move shared logic into components and on the screen components, leave only the logic of saving / receiving and callbacks for actions

vivalaakam commented on src/app.tsx:
```diff
@@ -292,6 +294,20 @@ export const App = () => {
+                    }}
+                  />
+                  <Stack.Screen
+                    name="settingsEditContact"
```
settingsContactEdit

vivalaakam commented on src/screens/settings-contact-edit.tsx:
```diff
@@ -0,0 +1,169 @@
+        textColorRight={LIGHT_GRAPHIC_GREEN_1}
+        textColorLeft={LIGHT_GRAPHIC_GREEN_1}
+      />
+      <KeyboardSafeArea>
```
move view logic into components please

vivalaakam commented on src/screens/transaction-contact-edit.tsx:
```diff
@@ -0,0 +1,80 @@
+  const cleanTextFile = () => setInputName('');
+
+  return (
+    <KeyboardSafeArea>
```
move view logic into components please

vivalaakam commented on src/screens/settings-account-edit.tsx:
```diff
@@ -0,0 +1,102 @@
+        textColorRight={LIGHT_GRAPHIC_GREEN_1}
+        textColorLeft={LIGHT_GRAPHIC_GREEN_1}
+      />
+      <KeyboardSafeArea>
```
move view logic into components

vivalaakam commented:
pleas, move all texts into i18n file =)

vivalaakam commented on src/components/actions-sheet.tsx:
```diff
@@ -0,0 +1,163 @@
+  }));
+
+  return (
+    <View style={page.container}>
```
```suggestion
    <View style={StyleSheet.absoluteFillObject}>
```

vivalaakam commented on src/components/actions-sheet.tsx:
```diff
@@ -0,0 +1,163 @@
+import React, {useEffect} from 'react';
```
how about to split logic into iOS and android files and in iOS use native realization?

vivalaakam commented on src/components/settings-address-book-edit.tsx:
```diff
@@ -0,0 +1,145 @@
+    const [inputName, setInputName] = useState(initName);
+
+    const onSubmit = () => {
+      if (isCreate) {
+        contacts.createContact(initAddress, inputName);
+      } else {
+        contacts.updateContact(initAddress, inputName);
+      }
```
these logic can be moved on top like

`   const onSubmit = () => {
  onSubmitFromProps(inputName)
       goBack();
   };
`

and in screen 

`
const onSubmit = (inputName) => {
     if (isCreate) {
         contacts.createContact(initAddress, inputName);
       } else {
         contacts.updateContact(initAddress, inputName);
       }
}
`

vivalaakam commented on src/components/settings-address-book-edit.tsx:
```diff
@@ -0,0 +1,145 @@
+            text: 'Delete',
+            style: 'destructive',
+            onPress: () => {
+              contacts.removeContact(initAddress);
```
move in screen logic

vivalaakam commented on src/hocs/with-actions-contact-item.tsx:
```diff
@@ -0,0 +1,108 @@
+        };
+        switch (screen) {
+          case 'transaction':
+            navigate('transactionContactEdit', params);
```
you can add here nextScreen prop and don't use switch

vivalaakam commented on src/i18n.ts:
```diff
@@ -49,5 +62,17 @@ const en: Record<I18N, string> = {
+  [I18N.settingsAccountEditHeaderTitle]: 'Edit account name',
+  [I18N.settingsContactEditHeaderTitle]: 'Contact',
+  [I18N.settingsContactEditDeleteContact]: 'Delete Contact',
+  [I18N.name]: 'Name',
+  [I18N.address]: 'Address',
+  [I18N.cancel]: 'Cancel',
+  [I18N.save]: 'Save',
+  [I18N.edit]: 'Edit',
+  [I18N.continue]: 'Continue',
```
please, use specific screen variable names

gHashTag replied:
Why make \``I18N.${screen}Address?`\`?
Address/name/save and others are simple words that can be used on many screens

vivalaakam commented on src/screens/home-settings.tsx:
```diff
@@ -38,7 +38,7 @@ export const HomeSettingsScreen = () => {
         title="Providers"
         next="settingsProviders"
-        style={page.button}
+        style={[page.button, page.hr]}
```
not following by design

vivalaakam commented on src/screens/settings-account-edit.tsx:
```diff
@@ -0,0 +1,101 @@
+import React, {useState} from 'react';
```
move view logic into components

gHashTag commented:
In the merge above, I added into Spacer component width and height props.
I also removed an unnecessary container
`src/screens/settings-address-book.tsx`
And add in `src/components/list-contact.tsx` bottom padding insets
